maestro-client/src/main/java/maestro/Filters.kt

```diff
@@ -209,4 +209,16 @@ object Filters {
         }
     }
 
+    fun deepestMatchingElement(filter: ElementFilter): ElementFilter {
```

Comment:
@amanjeetsingh150 @dmitry-zaitsev Won't this return duplicate nodes in the list since the initial list can contain both parent and child nodes?

Reply:
@amanjeetsingh150 Can you explain (1) what this filter is supposed to do and (2) why we need this?

Reply:
There were instances, especially in iOS, where we found that the accessibility like text was same for the element and its children. Earlier we used to select the first match which resulted in clicking the wrong element where should have clicked the deepest child. That would actually result in the click.